recipe/session/session_test.go

```diff
@@ -1215,6 +1219,145 @@ func TestGetSessionReturnsNilForRequestWithNoSessionWithCheckDatabaseTrueAndSess
 	}
 }
 
+type InfoLogData struct {
+	LastLine string
+	Output   []string
+}
+
+func GetInfoLogData(t *testing.T, startWith string) InfoLogData {
+	logFilePath := "../../../supertokens-root/logs/info.log"
+	file, err := os.Open(logFilePath)
+	if err != nil {
+		t.Fatalf("failed opening file: %s", err)
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	var lastLine string
+	var output []string
+
+	shouldRecordOutput := false
+
+	if startWith == "" {
+		shouldRecordOutput = true
+	}
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if strings.TrimSpace(line) != "" {
+			if startWith != "" && strings.Contains(line, startWith) {
+				shouldRecordOutput = true
+				continue
+			}
+
+			if shouldRecordOutput {
+				output = append(output, line)
+			}
+
+			lastLine = line
+		}
+	}
+
+	err = scanner.Err()
+	if err != nil {
+		t.Fatalf("scanner error: %s", err)
+	}
+
+	return InfoLogData{
+		LastLine: lastLine,
+		Output:   output,
+	}
+}
+
+func TestThatJWKSIsFetchedAsExpected(t *testing.T) {
+	originalRefreshlimit := JWKRefreshRateLimit
+	originalCacheAge := JWKCacheMaxAgeInMs
+
+	JWKRefreshRateLimit = 100
+	JWKCacheMaxAgeInMs = 2000
+
+	lastLineBeforeTest := GetInfoLogData(t, "").LastLine
+
+	configValue := supertokens.TypeInput{
+		Supertokens: &supertokens.ConnectionInfo{
+			ConnectionURI: "http://localhost:8080",
+		},
+		AppInfo: supertokens.AppInfo{
+			AppName:       "SuperTokens",
+			WebsiteDomain: "supertokens.io",
+			APIDomain:     "api.supertokens.io",
+		},
+		RecipeList: []supertokens.Recipe{
+			Init(nil),
+		},
+	}
+	BeforeEach()
+	unittesting.StartUpST("localhost", "8080")
+	defer AfterEach()
+	err := supertokens.Init(configValue)
+	if err != nil {
+		t.Error(err.Error())
+	}
+
+	mux := http.NewServeMux()
```

Comment:
you can use withoutrequestresponse versions of these functions and not have to use apis